cpp/arcticdb/version/version_map.hpp

```diff
@@ -152,17 +147,21 @@ class VersionMapImpl {
         auto next_key = ref_entry.head_;
         entry->head_ = ref_entry.head_;
         auto loaded_until = std::numeric_limits<VersionId>::max();
+        VersionId oldest_loaded_index_version = std::numeric_limits<VersionId>::max();
 
-        if ((load_params.load_type_ == LoadType::LOAD_LATEST || load_params.load_type_ == LoadType::LOAD_LATEST_UNDELETED ) && is_index_key_type(ref_entry.keys_[0].type())) {
+        if ((load_params.load_type_ == LoadType::LOAD_LATEST || load_params.load_type_ == LoadType::LOAD_LATEST_UNDELETED)
+            && is_index_key_type(ref_entry.keys_[0].type()))
+        {
             entry->keys_.push_back(ref_entry.keys_[0]);
         } else {
             do {
                 auto [key, seg] = store->read_sync(next_key.value());
                 std::tie(next_key, loaded_until) = read_segment_with_keys(seg, entry);
+                oldest_loaded_index_version = std::min(oldest_loaded_index_version, loaded_until);
```

Comment:
It should be safe to assume that the new loaded_until is always less than the previously-loaded one, but I don't object to the cautious approach

Reply:
That was my initial idea, but `read_segment_with_keys` returns valid version number only if an index was read. If for example a tombstone_all or tombstone is read it would return `std::numeric_limits<VersionId>::max`

Reply:
Aha. Cool, good to know